source/_integrations/azure_data_explorer.markdown

```diff
@@ -0,0 +1,186 @@
+
+Within a minute, you will have a Azure Data Explorer cluster ready
+
+After the creation of the database, copy the **Data ingestion URI** from the top of the page
+
+## Create Azure Data Table
+1. Navigate to [aka.ms/kustofree](https://aka.ms/kustofree).
+2. Navigate to **Query**.
+3. Write and execute the foloing statements one by one, replacing the content between the <> with the copied values (including the brackets)
+
+```KQL
+// Give the Service Pricipal write access to the database
+.add database ['<databasename>'] ingestors ('aadapp=<ApplicationID>;<DirectoryID>');
+
+// Give the Service Pricipal read access to database (used for connectivity checks) 
+.add database ['<databasename>'] viewers ('aadapp=<ApplicationID>;<DirectoryID>');
+
+// Create a table for the data to be ingested into (Replace name and copy inserted *name* for later use)
+.create table ['<name_to_be_replaced>'] (entity_id: string, state: string, attributes: dynamic, last_changed: datetime, last_updated: datetime, context: dynamic)
+
+// Creat a mapping from the incomming JSON to the table and collums just created (replace name with table name from previous step)
+.create table ['<name_to_be_replaced>'] ingestion json mapping ['ha_json_mapping'] '[{"column":"entity_id","path":"$.entity_id"},{"column":"state","path":"$.state"},{"column":"attributes","path":"$.attributes"},{"column":"last_changed","path":"$.last_canged"},{"column":"last_updated","path":"$.last_updated"},{"column":"context","path":"$.context"}]'
+```
+
+This is an example with a free cluster for reference
+
+```KQL
+.add database ['HomeAssistant'] ingestors ('aadapp=b5253d02-c8f4-1234-a0f0-818491ba2a1f;72f123bf-86f1-41af-91ab-2d7cd011db93');
+
+.add database ['HomeAssistant'] viewers ('aadapp=b5253d02-c8f4-1234-a0f0-818491ba2a1f;72f123bf-86f1-41af-91ab-2d7cd011db93');
+
+.create table ['raw'] (entity_id: string, state: string, attributes: dynamic, last_changed: datetime, last_updated: datetime, context: dynamic)
+
+.create table ['raw'] ingestion json mapping ['ha_json_mapping'] '[{"column":"entity_id","path":"$.entity_id"},{"column":"state","path":"$.state"},{"column":"attributes","path":"$.attributes"},{"column":"last_changed","path":"$.last_canged"},{"column":"last_updated","path":"$.last_updated"},{"column":"context","path":"$.context"}]'
+```
+
+## Configuration
+>if using a free cluste, check the **Use Queueing client** in the form
+
+{% include integrations/config_flow.md %}
+
+After completiing the flow, Home Assistant is sending data to Azure Data Explorer. 
+
+> Home Assistant is buffering for defualt 5 seconds before sending, and Batching Policy in Azure Data Explorer will futher batch up for default 
+
+## Filters
+
+Optinaly add the following lines to your `configuration.yaml` file for filtering data ingested into Azure Data Explorer:
+
+```yaml
+# Example configuration.yaml entry
+azure_data_explorer:
+  filter:
+    include_domains:
+    - homeassistant
+    - light
+    - media_player
+```
```

Comment:
Why is this not handled in an options_flow in core but in YAML? And the first setup of the integration go through the config_flow? I think it's still a point of discussion in the core pull request.

Reply:
@kaareseras @klaasnicolaas Is this a blocking change? Could you please guide on the next steps?

Reply:
No, this is not blocking. Still next step is for someone to pick this up for review and release

Reply:
@frenck Request your review.